### ISSUE TITLE
Add command for `Quarto: Set Editor Default`

### ISSUE THREAD

**Issue by DavisVaughan** [quarto-dev/quarto#596]
Right now you can _globally_ tell VS Code / Positron to use the visual editor by default for ALL qmd documents with this setting:

```json
    "workbench.editorAssociations": {
        "*.qmd": "quarto.visualEditor"
    }
```

This is officially supported as the way for a user to set their global preference. It's the reason we set `priority: "option"` for our custom editor in `package.json`

```
    "customEditors": [
      {
        "viewType": "quarto.visualEditor",
        "displayName": "Quarto Visual Editor",
        "selector": [
          {
            "filenamePattern": "*.{md,qmd}"
          }
        ],
        "priority": "option"
      }
    ],
```

https://code.visualstudio.com/api/references/contribution-points#contributes.customEditors

![Image](https://github.com/user-attachments/assets/184d7f30-4a19-422a-b46d-af430a42a370)

i.e. we don't forcibly take over qmds and make them open in visual mode, but a user can use this editor association option to opt in to this global override if they always want visual mode.

---

This is a super cool feature, but it's super hard to find!

I vote we make an ephemeral _command_ that the user can call as something like `Quarto: Set Editor Default` who's sole job would be to follow up with a quick pick where the user can select either `Source` or `Visual`, and then based on their answer we set this `editorAssociation` option for them. It's possible we should also ask them if they want it set at a `User` or `Workspace` level.

---

Note that I do not think we should make this a quarto level _option_. That leaves the door open for the quarto option to get out of sync with the editor association option, like with manual editing of settings you can confusingly end up with this if we did an option:

```json
    "quarto.editorDefault = "source",
    "workbench.editorAssociations": {
        "*.qmd": "quarto.visualEditor"
    }
```

and that seems bad. So having it as a command that just sets `editorAssociations` seems like a better way to do this.

**Comment by juliasilge** [comment]
This sort of already exists, since the Quarto extension provides the visual editor editor with ` "priority": "option"`:

![Image](https://github.com/user-attachments/assets/e0a26d7c-323c-4b49-99f6-631ca1768b78)

This quick pick pops up if you ever right click on a `.qmd` to open it and choose "Open With...". I looked a little bit at whether we can hook a command up to get to this spot but don't see it quite yet.

**Comment by juliasilge** [comment]
This quick pick comes from `doPickEditor()` with `showDefaultPicker` set to true:

https://github.com/posit-dev/positron/blob/8d0c666584e6a17ade85e6e97e7b9cb6df1eba30/src/vs/workbench/services/editor/browser/editorResolverService.ts#L782-L783

It doesn't look to me like any of that is exposed to extensions, for extensions to prompt this.

**Comment by DavisVaughan** [comment]
Oh nice yea if you hit "configure default" after `Open with...` then if you set it to visual editor then i do see this in my settings.json now

```
    "workbench.editorAssociations": {
        "*.qmd": "quarto.visualEditor"
    }
```

That's basically what I'm asking for, just as a slightly simplified version that would only show `Source` and `Visual` as options, and is more easily discovered through being a quarto command under `> Quarto: Set Editor Default` in the command palette 

It wouldn't have to use exactly the same UI, so we would not need to access that existing quick picker directly. We'd just make our own simple one that only shows `Source` and `Visual` as options.

Or we can decide the current approach is good enough, if a bit hidden

**Comment by gregmacfarlane** [comment]
I came across this post looking desperately for a way to *stop* using the visual editor. Every time I open a .qmd file the visual editor re-writes all of my line endings and latex formatting. Still haven't found it yet, because the dropdown that @juliasilge pointed out still takes me to the visual editor no matter which option I pick!

**Comment by juliasilge** [comment]
Hmmmm, interesting. @gregmacfarlane can you walk through these steps, in either VS Code or Positron?

- In the File Explorer, right click on any `.qmd` file and choose "Open With..."

![Image](https://github.com/user-attachments/assets/20283116-f01f-40a2-bd6b-503f78c41892)

- Click on "Configure default editor for '*.qmd'"

![Image](https://github.com/user-attachments/assets/1a444f60-b8a3-4d07-b573-e08e23805aa2)

- Choose "Text Editor"

![Image](https://github.com/user-attachments/assets/b10c83c3-c4d8-4ef4-af4a-3c0648ff655a)

That will set this in your user settings:

```json
"workbench.editorAssociations": {
        "*.qmd": "default"
    }
```

Alternatively, you can manually set this in your user `settings.json`.

Does this not work for you?

**Comment by juliasilge** [comment]
Apologies if you have already checked these, but can you check:

- do you have `editor: visual` in your `_quarto.yml` for this project or your specific file?
- does the specific file `financial.qmd` set [any `editor` options](https://quarto.org/docs/visual-editor/vscode/#markdown-output) in it?
- what happens if you make a brand new simple `.qmd` file in the same project as this file? does it open in source or visual mode?

**Comment by gregmacfarlane** [comment]
Turns out the issue was in the `editor: visual` in the project `_quarto.yml` file. It's interesting that this overrides all of the other things, but i've got it now.

**Comment by juliasilge** [comment]
Yep, there are hierarchical rules such that a document-level setting overrides a project-level setting which overrides a user-level setting. The motivation here is, for example, that you may collaborate with someone on a project and you both will use the same settings on the project, no matter what individual user-level settings you have.
